Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/VehicleRegistrationCertificate.hs

```diff
@@ -229,9 +250,91 @@ onVerifyRC verificationReq output = do
+  Esq.runNoTransaction $ DAQuery.updateIsRcActiveByRcAndDriver pId rc.id
+  Esq.runNoTransaction $ VQuery.deleteById pId
+  return Success
+
+validateCanRcActive :: (Id Person.Person, Id DM.Merchant) -> Text -> Flow APISuccess
+validateCanRcActive (driverId, merchantId) vehicleRegistrationCertNumber = do
+  rc <- RCQuery.findLastVehicleRC vehicleRegistrationCertNumber >>= fromMaybeM (InvalidRequest "Rc not found")
+  activeRC' <- DAQuery.findByRcIdIsRcActive rc.id
+  case activeRC' of
+    Just activeRC -> do
+      isOnRide <- DIQuery.findByDriverIdActiveRide (cast activeRC.driverId)
+      when (isJust isOnRide) $ throwError (InvalidRequest "Vehicle on ride. try again later.")
+      lastRideAssignedOn <- RQuery.findLastRideAssigned activeRC.driverId >>= fromMaybeM (PersonNotFound activeRC.driverId.getId)
+      now <- getCurrentTime
+      if abs (diffUTCTime now lastRideAssignedOn.createdAt) > 5 * 24 * 60 * 60
+        then do
+          Esq.runNoTransaction $ DAQuery.findByDriverIdAndUpdateIsRcActive lastRideAssignedOn.driverId
+          let vehicle = buildVehicle now driverId merchantId vehicleRegistrationCertNumber rc
+          Esq.runTransaction $ VQuery.upsert vehicle
+          makeRcInactive (driverId, merchantId) vehicleRegistrationCertNumber
+          return Success
+        else throwError $ InvalidRequest "Rc Active on another driver account"
+    Nothing -> do
+      makeRcInactive (driverId, merchantId) vehicleRegistrationCertNumber
+      now <- getCurrentTime
+      let vehicle = buildVehicle now driverId merchantId vehicleRegistrationCertNumber rc
+      Esq.runTransaction $ VQuery.upsert vehicle
+      Esq.runTransaction $ DAQuery.updateIsRcAndDriverIdActiveByRc rc.id driverId
+      return Success
+  where
+    buildVehicle now personId_ merchantId_ certificateNumber rc =
+      Vehicle.Vehicle
+        { Vehicle.driverId = personId_,
+          Vehicle.capacity = rc.vehicleCapacity,
+          Vehicle.category = Vehicle.getCategory <$> rc.vehicleVariant,
+          Vehicle.make = rc.vehicleManufacturer,
+          Vehicle.model = fromMaybe "Unkown" rc.vehicleModel,
+          Vehicle.size = Nothing,
+          Vehicle.merchantId = merchantId_,
+          Vehicle.variant = fromMaybe AUTO_RICKSHAW rc.vehicleVariant, -- Value will be always Just if reaching here
+          Vehicle.color = fromMaybe "Unkown" rc.vehicleColor,
+          Vehicle.energyType = rc.vehicleEnergyType,
+          Vehicle.registrationNo = certificateNumber,
+          Vehicle.registrationCategory = Nothing,
+          Vehicle.vehicleClass = fromMaybe "Unkown" rc.vehicleClass,
+          Vehicle.createdAt = now,
+          Vehicle.updatedAt = now
+        }
+
+deleteRc :: (Id Person.Person, Id DM.Merchant) -> DeleteRcReq -> Flow APISuccess
+deleteRc (driverId, _) DeleteRcReq {..} = do
+  rc <- RCQuery.findLastVehicleRC rcNo >>= fromMaybeM (InvalidRequest "Rc not found")
+  Esq.runNoTransaction $ DAQuery.deleteByRcAndDriverId driverId rc.id
```

Comment:
Shouldn't you check for if RC is already on ride and all those things before deleting it. And why to delete, when we can just add one flag `isDeleted` inside DriverRCAssociation toggle that only.